### DISCUSSION TITLE
How to make visitor like in ANTLR?

### DISCUSSION THREAD

**Discussion by yaskhan** [tree-sitter/tree-sitter#860] [Q&A] [answered]
Hello. How to make visitor like in ANTLR? Give an example if you can. Thanx.
(JavaScript)

**Comment by marcel0ll** [comment, marked as answer]
Hey @yaskhan it's not the same as ANTLR, but I am using node type from ts_node_type(node) to call some functions stored in a hashmap on node enter and on node exit. I am building a javascript minifier on top of it to test it out. 

[tree-sitter-visitor](https://github.com/marcel0ll/tree-sitter-visitor)

[javascript minifier](https://github.com/marcel0ll/jsminify)

Hope it helps
